src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -1243,6 +1221,88 @@ lExit: ;
 }
 FCIMPLEND
 
+static FC_BOOL_RET IsFastPathSupportedHelper(FieldDesc* pFieldDesc)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        PRECONDITION(CheckPointer(pFieldDesc));
+    }
+    CONTRACTL_END;
+
+    return pFieldDesc->IsThreadStatic() ||
+        pFieldDesc->IsEnCNew() ||
+        pFieldDesc->IsCollectible() ? FALSE : TRUE;
+}
+
+FCIMPL1(FC_BOOL_RET, RuntimeFieldHandle::IsFastPathSupported, ReflectFieldObject *pFieldUNSAFE)
+{
+    CONTRACTL {
+        FCALL_CHECK;
+    }
+    CONTRACTL_END;
+
+    REFLECTFIELDREF refField = (REFLECTFIELDREF)ObjectToOBJECTREF(pFieldUNSAFE);
+    if (refField == NULL)
+        FCThrowRes(kArgumentNullException, W("Arg_InvalidHandle"));
```

Comment:
Could you please move all new FCThrows to the managed side? We are trying to get rid of all of them.

(This side can have assert.)

Reply:
I remove them; an assert here is fine since we don't expect rogue callers.